diff --git a/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt b/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt
index 43baf2a..e36e5f5 100644
--- a/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt
+++ b/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt
@@ -6,7 +6,8 @@ build_sub22_mests.cc  feedframe.cc          mblock_bsumsq_mmx.cc        motionse
 build_sub44_mests.cc  find_best_one_pel.cc  mblock_sad_mmx.cc           mpeg2enc.cc      predict_mmx.cc    putvlc.cc        synchrolib.cc
 conform.cc            idct.cc               mblock_sad_mmxe.cc          mpeg2enc_if.cc   predict_ref.cc    quantize.cc      transfrm.cc
 cpu_accel.cc          idctdata.cc           mblock_sub44_sads_x86.cc    mpeg2param.cc    predict_x86.cc    quantize_ref.cc  transfrm_ref.cc
-fdctdata.cc           idct_mmx.cc           mjpeg_logging.c				mpeg2setup.cc    putbits.cc        quantize_x86.cc  transfrm_x86.cc
+fdctdata.cc           idct_mmx.cc           
+mpeg2setup.cc         putbits.cc            quantize_x86.cc             transfrm_x86.cc
 fdct_mmx.cc           ioio.c                mblock_sumsq_mmx.cc         mpegconsts.cc    puthdr.cc         quant_mmx2.cc    writepic.cc
 fdctref.cc            macroblock.cc         motion.cc                   predcomp_mmx.cc  putmpg.cc         ratectl.cc)
 
@@ -45,4 +46,5 @@ add_library(${ADM_LIB} ${${ADM_LIB}_SRCS})
 
 if (UNIX)
 	add_target_cflags(mpeg2enc -fPIC)
-endif (UNIX)
\ No newline at end of file
+endif (UNIX)
+
