--- avidemux_2.4.4.orig/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp	2008-09-23 21:09:49.000000000 -0400
+++ avidemux_2.4.4/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp	2009-02-19 23:05:30.000000000 -0500
@@ -60,7 +60,7 @@
 {
 	printf("\n[Locale] Locale: %s\n", QLocale::system().name().toUtf8().constData());
 
-	QString appdir = QCoreApplication::applicationDirPath() + "/i18n/";
+	QString appdir = "/usr/share/avidemux/i18n/";
 
 	loadTranslation(&qtTranslator, appdir + "qt_" + QLocale::system().name());
 	loadTranslation(&avidemuxTranslator, appdir + "avidemux_" + QLocale::system().name());
--- avidemux_2.4.4.orig/cmake/Ts.cmake	2007-10-14 07:58:28.000000000 -0400
+++ avidemux_2.4.4/cmake/Ts.cmake	2009-02-19 23:04:55.000000000 -0500
@@ -63,7 +63,7 @@
                 
             SET(qm_files ${qm_files} ${_outXml} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
@@ -94,7 +94,7 @@
                 
             SET(qm_files ${qm_files} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
