diff -ur avidemux_2.5.3/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp avidemux_2.5.3-ffmpeg-aac/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
--- avidemux_2.5.3/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp	2010-02-23 19:21:06.000000000 +0100
+++ avidemux_2.5.3-ffmpeg-aac/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp	2010-10-02 03:34:42.000000000 +0200
@@ -29,13 +29,16 @@
 
 
 #define Join(x,y) x##_##y
-#if defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3)
+#if defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3) && !defined(ADM_LAV_AAC)
   #define makeName(x) Join(x,MP2)
   #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_MP2
 #else
- #if !defined(ADM_LAV_MP2) && defined(ADM_LAV_AC3)
+ #if !defined(ADM_LAV_MP2) && defined(ADM_LAV_AC3) && !defined(ADM_LAV_AAC)
   #define makeName(x) Join(x,AC3)
   #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_AC3
+ #elif !defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3) && defined(ADM_LAV_AAC)
+  #define makeName(x) Join(x,AAC)
+  #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_AAC
  #else
    #error
  #endif
@@ -64,13 +67,19 @@
   2,                    // Max channels
   1,0,0,                // Version
 #else
-  
-
+#ifdef ADM_LAV_AC3
  "LavAC3",            
   "AC3 (lav)",      
   "AC3 LavEncoder encoder plugin Mean 2008",             
   6,                    // Max channels
   1,0,0,                // Version
+#else
+ "LavAAC",
+  "AAC (lav)",
+  "AAC LavEncoder encoder plugin Mean 2008",
+  6,                    // Max channels
+  1,0,0,                // Version
+#endif
 #endif
   makeName(WAV),
 
@@ -136,7 +145,9 @@
 #endif
   _wavheader->byterate=(lavConfig.bitrate*1000)>>3;         
       
-#ifdef ADM_LAV_MP2 
+#if defined(ADM_LAV_AAC)
+    _chunk = 1024*_wavheader->channels;
+#elif defined(ADM_LAV_MP2)
     _chunk = 1152*_wavheader->channels;
 #else
     _chunk = 1536*_wavheader->channels; // AC3
diff -ur avidemux_2.5.3/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt avidemux_2.5.3-ffmpeg-aac/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt
--- avidemux_2.5.3/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt	2010-02-28 21:32:28.000000000 +0100
+++ avidemux_2.5.3-ffmpeg-aac/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt	2010-10-02 03:30:48.000000000 +0200
@@ -27,3 +27,12 @@
 
 	INIT_AUDIO_ENCODER(ADM_ae_lav_ac3)
 	INSTALL_AUDIOENCODER(ADM_ae_lav_ac3)
+
+	SET(ADM_ae_lav_aac_SRCS audioencoder_lavcodec.cpp)
+
+	ADD_LIBRARY(ADM_ae_lav_aac SHARED ${ADM_ae_lav_aac_SRCS})
+	ADD_TARGET_CFLAGS(ADM_ae_lav_aac "-DADM_LAV_AAC")
+	TARGET_LINK_LIBRARIES(ADM_ae_lav_aac ADM_core ADM_coreAudio ADM_coreUI ADM_libavcodec ADM_libavutil)
+
+	INIT_AUDIO_ENCODER(ADM_ae_lav_aac)
+	INSTALL_AUDIOENCODER(ADM_ae_lav_aac)

