--- avidemux/CMakeLists.txt.tmplinktarget	2009-10-01 15:14:16.000000000 -0400
+++ avidemux/CMakeLists.txt	2009-10-10 19:13:17.500150360 -0400
@@ -503,15 +503,6 @@
 	ENDIF (MAC_BUNDLE_DIR)
 ELSE (APPLE)
 	INSTALL(TARGETS ${ADM_EXES} RUNTIME DESTINATION ${BIN_DIR} LIBRARY DESTINATION lib${LIB_SUFFIX})
-ENDIF (APPLE)
-
-IF (UNIX AND NOT APPLE)
-   ADD_CUSTOM_COMMAND(TARGET ffmpeg
-                      POST_BUILD
-                      COMMAND mkdir -p ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
-                      COMMAND find ${CMAKE_BINARY_DIR}/avidemux -name '*.so*' | xargs ln -sft ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
-                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-                      COMMENT "Creating temporary lib directory for proper plugins linking")
-ENDIF (UNIX AND NOT APPLE)
+ENDIF (APPLE)
 
 
