diff -p -up avidemux_2.4.1/configure.in.in~ avidemux_2.4.1/configure.in.in
--- avidemux_2.4.1/configure.in.in~	2008-02-13 20:04:21.000000000 +0100
+++ avidemux_2.4.1/configure.in.in	2008-02-17 21:43:38.000000000 +0100
@@ -195,7 +195,6 @@ adm_qt4_enabled=yes
 echo " ------------------- Probing for QT 4.x ------------------"
 QTDIR=xxx
 
-QTPOSTFIX="-qt4"
 QTEXTRAINC="/usr/include/qt4"
 QTEXTRALIB="/usr/lib/qt4"
 
