diff -Naur avidemux_2.6.10.orig/cmake/Ts.cmake avidemux_2.6.10/cmake/Ts.cmake
--- avidemux_2.6.10.orig/cmake/Ts.cmake	2015-06-10 00:42:47.000000000 -0500
+++ avidemux_2.6.10/cmake/Ts.cmake	2015-11-04 15:22:23.336615398 -0600
@@ -15,24 +15,17 @@
 
 MACRO(FIND_LRELEASE)
     IF(NOT LRELEASE_EXECUTABLE AND NOT LRELEASE_NOT_FOUND)
-		FIND_PROGRAM(LRELEASE_EXECUTABLE lrelease PATHS
-			"[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\4.0.0;InstallDir]/bin"
-			"[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/bin"
-			$ENV{QTDIR}/bin)
-
-		IF (NOT LRELEASE_EXECUTABLE) # search again under the name lrelease-qt4
-        		FIND_PROGRAM(LRELEASE_EXECUTABLE lrelease-qt4 PATHS
-			$ENV{QTDIR}/bin)
-		        IF (NOT LRELEASE_EXECUTABLE) # search again under the name lrelease-qt4
-			        MESSAGE(FATAL_ERROR "${LRELEASE_EXECUTABLE} not found - ts files can't be processed")
-			        SET(LRELEASE_NOT_FOUND "1")     # to avoid double checking in one cmake run
-		        ENDIF (NOT LRELEASE_EXECUTABLE) # search again under the name lrelease-qt4
-		ENDIF (NOT LRELEASE_EXECUTABLE)
-		IF (LRELEASE_EXECUTABLE) 
-                        MESSAGE(STATUS "lrelease found as ${LRELEASE_EXECUTABLE}")
-		ENDIF (LRELEASE_EXECUTABLE) 
-
-	ENDIF(NOT LRELEASE_EXECUTABLE AND NOT LRELEASE_NOT_FOUND)
+	FIND_PROGRAM(LRELEASE_EXECUTABLE NAMES lrelease lrelease-qt4 lrelease-qt5
+            PATHS
+		"[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\4.0.0;InstallDir]/bin"
+		"[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/bin"
+		$ENV{QTDIR}/bin)
+	IF (LRELEASE_EXECUTABLE) 
+            MESSAGE(STATUS "lrelease found as ${LRELEASE_EXECUTABLE}")
+        ELSE (LRELEASE_EXECUTABLE)
+            MESSAGE(FATAL_ERROR "lrelease not found - ts files can't be processed")
+	ENDIF (LRELEASE_EXECUTABLE) 
+    ENDIF(NOT LRELEASE_EXECUTABLE AND NOT LRELEASE_NOT_FOUND)
 ENDMACRO(FIND_LRELEASE)
 #
 #
