--- avidemux_2.5.0/avidemux/ADM_core/include/ADM_misc.h.ark	2009-06-21 16:50:12.412942889 +0200
+++ avidemux_2.5.0/avidemux/ADM_core/include/ADM_misc.h	2009-06-21 16:50:18.760943176 +0200
@@ -10,7 +10,7 @@
 #define ADM_MISC_H
 
 #include "ADM_coreConfig.h"
-
+#include <stdint.h>
 
 typedef struct 
 {
--- avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp.ark	2009-06-21 16:35:38.140135807 +0200
+++ avidemux_2.5.0/avidemux/ADM_core/src/ADM_fileio.cpp	2009-06-21 16:35:51.867969517 +0200
@@ -681,7 +681,7 @@ void ADM_PathStripName(char *str)
 */
 const char *ADM_GetFileName(const char *str)
 {
-	char *filename;
+	const char *filename;
 	char *filename2;
 
 #ifndef __WIN32
--- avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_coreToolkit.h.ark	2009-06-21 16:57:14.895092062 +0200
+++ avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_coreToolkit.h	2009-06-21 16:57:23.752111783 +0200
@@ -19,6 +19,8 @@
 #ifndef ADM_COREUI_TOOLKIT__
 #define ADM_COREUI_TOOLKIT__
 
+#include <stdint.h>
+
 typedef enum 
 {
         ADM_LOG_NONE=0,
--- avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_enter.h.ark	2009-06-21 16:55:46.188973629 +0200
+++ avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_enter.h	2009-06-21 16:55:53.465409726 +0200
@@ -15,6 +15,7 @@
  ***************************************************************************/
 #ifndef DIA_ENTER_H
 #define DIA_ENTER_H
+#include <stdint.h>
 
 //  Get a float value
 uint8_t  DIA_GetFloatValue(float *value, float min, float max, const char *title, const char *legend);
--- avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_fileSel.h.ark	2009-06-21 16:55:31.471073751 +0200
+++ avidemux_2.5.0/avidemux/ADM_coreUI/include/DIA_fileSel.h	2009-06-21 16:55:39.379067926 +0200
@@ -15,6 +15,7 @@
 
 #ifndef DIA_FILESEL_H
 #define DIA_FILESEL_H
+#include <stdint.h>
 typedef void SELFILE_CB(const char *);
 
 // Associated functions type
--- avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp.ark	2009-06-21 16:40:22.066196323 +0200
+++ avidemux_2.5.0/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp	2009-06-21 16:40:29.289001011 +0200
@@ -71,7 +71,7 @@ fileParser::~fileParser()
 */
 uint8_t fileParser::open( const char *filename,FP_TYPE *multi )
 {
-        char *dot = NULL;                   // pointer to the last dot in filename
+        const char *dot = NULL;                   // pointer to the last dot in filename
         uint8_t decimals = 0;               // number of decimals
         char *left = NULL, *number = NULL, *right = NULL; // parts of filename (after splitting)
 
--- avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc.ark	2009-06-21 16:42:15.574312778 +0200
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc	2009-06-21 16:42:18.926951396 +0200
@@ -104,7 +104,7 @@ void y4m_ratio_reduce(y4m_ratio_t *r)
 
 int y4m_parse_ratio(y4m_ratio_t *r, const char *s)
 {
-  char *t = strchr(s, ':');
+  const char *t = strchr(s, ':');
   if (t == NULL) return Y4M_ERR_RANGE;
   r->n = atoi(s);
   r->d = atoi(t+1);
--- avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp.ark	2009-06-21 16:43:15.988362511 +0200
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp	2009-06-21 16:43:19.560943520 +0200
@@ -53,7 +53,7 @@ LPCMStream::LPCMStream(IBitStream &ibs, 
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
--- avidemux_2.5.0/avidemux/ADM_UIs/ADM_QT4/src/T_threadCount.h.ark	2009-06-21 16:48:26.935280369 +0200
+++ avidemux_2.5.0/avidemux/ADM_UIs/ADM_QT4/src/T_threadCount.h	2009-06-21 16:48:38.041943420 +0200
@@ -7,6 +7,7 @@
 #include <QtGui/QRadioButton>
 #include <QtGui/QSpinBox>
 #include <QtGui/QWidget>
+#include <stdint.h>
 
 namespace ADM_qt4Factory
 {
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_commonUI/ADM_tray.h.ark	2009-06-21 17:06:07.032249029 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_commonUI/ADM_tray.h	2009-06-21 17:06:13.612069717 +0200
@@ -17,6 +17,7 @@
 
 #ifndef ADM_TRAY_H
 #define ADM_TRAY_H
+#include <stdint.h>
 
 class ADM_tray
 {
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_color.cpp.ark	2009-06-21 17:00:29.869950030 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/DIA_color.cpp	2009-06-21 17:00:36.084898982 +0200
@@ -1,4 +1,5 @@
 #include <QtGui/QColorDialog>
+#include <stdint.h>
 
 #include "ADM_toolkitQt.h"
 
@@ -17,4 +18,4 @@ int DIA_colorSel(uint8_t *r, uint8_t *g,
 	}
 
 	return 0;
-}
\ No newline at end of file
+}
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_jobs.cpp.ark	2009-06-21 16:54:38.555192979 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_jobs.cpp	2009-06-21 16:54:44.176068089 +0200
@@ -6,6 +6,7 @@
 #include "Q_jobs.h"
 #include "DIA_coreToolkit.h"
 #include "ADM_toolkitQt.h"
+#include <cstdio>
 
 static void updateStatus(void);
 extern bool parseECMAScript(const char *name);
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_license.cpp.ark	2009-06-21 16:53:15.590997514 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_dialog/Q_license.cpp	2009-06-21 16:53:23.719993769 +0200
@@ -17,6 +17,7 @@
 
 #include "Q_license.h"
 #include "ADM_toolkitQt.h"
+#include <stdint.h>
 
 Ui_licenseWindow::Ui_licenseWindow(QWidget *parent) : QDialog(parent)
 {
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp.ark	2009-06-21 17:04:44.617943090 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp	2009-06-21 17:04:50.589943463 +0200
@@ -14,6 +14,7 @@ Custom slider
 
 #include <QtGui/QPainter>
 #include <QtGui/QSlider>
+#include <cstdio>
 
 #include "ADM_qslider.h"
 
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h.ark	2009-06-21 17:01:19.298477728 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h	2009-06-21 17:01:25.233500616 +0200
@@ -1,5 +1,6 @@
 #ifndef ADM_Q_SLIDER_H
 #define ADM_Q_SLIDER_H
+#include <stdint.h>
 
 class ADM_QSlider : public QSlider
 {
--- avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp.ark	2009-06-21 17:04:07.237197461 +0200
+++ avidemux_2.5.0/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp	2009-06-21 17:04:14.105432535 +0200
@@ -9,6 +9,7 @@
 //
 
 #include "config.h"
+#include <stdint.h>
 
 #include <QtGui/QApplication>
 #include <QtGui/QDesktopWidget>
