diff -up avidemux_2.4.3/avidemux/ADM_colorspace/ADM_rgb.h.compile avidemux_2.4.3/avidemux/ADM_colorspace/ADM_rgb.h
--- avidemux_2.4.3/avidemux/ADM_colorspace/ADM_rgb.h.compile	2008-07-20 09:19:20.000000000 +0200
+++ avidemux_2.4.3/avidemux/ADM_colorspace/ADM_rgb.h	2009-03-22 14:19:00.000000000 +0100
@@ -19,6 +19,8 @@
  #ifndef COLYUVRGB_H
  #define COLYUVRGB_H
 
+#include <inttypes.h>
+
 #define ADM_COLOR_IS_YUV 0x1000
  typedef enum 
  {
diff -up avidemux_2.4.3/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp.compile avidemux_2.4.3/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp
--- avidemux_2.4.3/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp.compile	2008-01-11 14:37:18.000000000 +0100
+++ avidemux_2.4.3/avidemux/ADM_inputs/ADM_mpegdemuxer/dmx_io.cpp	2009-03-22 13:53:05.000000000 +0100
@@ -71,7 +71,7 @@ fileParser::~fileParser()
 */
 uint8_t fileParser::open( const char *filename,FP_TYPE *multi )
 {
-        char *dot = NULL;                   // pointer to the last dot in filename
+        const char *dot = NULL;                   // pointer to the last dot in filename
         uint8_t decimals = 0;               // number of decimals
         char *left = NULL, *number = NULL, *right = NULL; // parts of filename (after splitting)
 
diff -up avidemux_2.4.3/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc.compile avidemux_2.4.3/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc
--- avidemux_2.4.3/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc.compile	2004-01-30 20:32:00.000000000 +0100
+++ avidemux_2.4.3/avidemux/ADM_libraries/ADM_libmpeg2enc/yuv4mpeg_ratio.cc	2009-03-22 14:00:34.000000000 +0100
@@ -104,7 +104,7 @@ void y4m_ratio_reduce(y4m_ratio_t *r)
 
 int y4m_parse_ratio(y4m_ratio_t *r, const char *s)
 {
-  char *t = strchr(s, ':');
+  const char *t = strchr(s, ':');
   if (t == NULL) return Y4M_ERR_RANGE;
   r->n = atoi(s);
   r->d = atoi(t+1);
diff -up avidemux_2.4.3/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp.compile avidemux_2.4.3/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp
--- avidemux_2.4.3/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp.compile	2007-08-03 02:04:40.000000000 +0200
+++ avidemux_2.4.3/avidemux/ADM_libraries/ADM_mplex/lpcmstrm_in.cpp	2009-03-22 14:01:28.000000000 +0100
@@ -53,7 +53,7 @@ LPCMStream::LPCMStream(IBitStream &ibs, 
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
diff -up avidemux_2.4.3/avidemux/ADM_osSupport/ADM_fileio.cpp.compile avidemux_2.4.3/avidemux/ADM_osSupport/ADM_fileio.cpp
--- avidemux_2.4.3/avidemux/ADM_osSupport/ADM_fileio.cpp.compile	2008-06-29 00:34:26.000000000 +0200
+++ avidemux_2.4.3/avidemux/ADM_osSupport/ADM_fileio.cpp	2009-03-22 13:52:03.000000000 +0100
@@ -515,7 +515,7 @@ void		PathStripName(char *str)
 */
 const char *GetFileName(const char *str)
 {
-	char *filename;
+	const char *filename;
         char *filename2;
 #ifndef ADM_WIN32		
 	filename = strrchr(str, '/');
diff -up avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp.compile avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp
--- avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp.compile	2009-03-22 14:10:27.000000000 +0100
+++ avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.cpp	2009-03-22 14:15:51.000000000 +0100
@@ -13,7 +13,7 @@ Custom slider
 ***************************************************************************/
 
 #include "config.h"
-
+#include <cstdio>
 #include <math.h>
 #include <QPainter>
 #include <QSlider>
diff -up avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h.compile avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h
--- avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h.compile	2007-08-21 17:02:54.000000000 +0200
+++ avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/ADM_qslider.h	2009-03-22 14:16:01.000000000 +0100
@@ -1,6 +1,6 @@
 #ifndef ADM_Q_SLIDER_H
 #define ADM_Q_SLIDER_H
-
+#include <inttypes.h>
 class ADM_QSlider : public QSlider
 {
 protected:
diff -up avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp.compile avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp
--- avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp.compile	2008-03-06 19:48:52.000000000 +0100
+++ avidemux_2.4.3/avidemux/ADM_userInterfaces/ADM_QT4/ADM_gui/gui_none.cpp	2009-03-22 14:09:29.000000000 +0100
@@ -9,7 +9,7 @@
 //
 
 #include "config.h"
-
+#include <inttypes.h>
 #include <QtGui/QApplication>
 #include <QtGui/QDesktopWidget>
 
diff -up avidemux_2.4.3/avidemux/ADM_video/mmx_macros.h.compile avidemux_2.4.3/avidemux/ADM_video/mmx_macros.h
--- avidemux_2.4.3/avidemux/ADM_video/mmx_macros.h.compile	2006-10-31 16:35:20.000000000 +0100
+++ avidemux_2.4.3/avidemux/ADM_video/mmx_macros.h	2009-03-22 13:47:06.000000000 +0100
@@ -353,7 +353,7 @@ mmx_ok(void)
 #define	mmx_m2r(op, mem, reg) \
 	__asm__ __volatile__ (#op " %0, %%" #reg \
 			      : /* nothing */ \
-			      : "X" (mem))
+			      : "m" (mem))
 
 #define	mmx_r2m(op, reg, mem) \
 	__asm__ __volatile__ (#op " %%" #reg ", %0" \
