diff -up avidemux_2.4.4/avidemux/CMakeLists.txt.libdca avidemux_2.4.4/avidemux/CMakeLists.txt
--- avidemux_2.4.4/avidemux/CMakeLists.txt.libdca	2009-03-23 23:08:35.000000000 +0100
+++ avidemux_2.4.4/avidemux/CMakeLists.txt	2009-03-23 23:10:17.000000000 +0100
@@ -315,7 +315,11 @@ endif(USE_AMR_NB AND NOT USE_LATE_BINDIN
 
 #libdts
 if(USE_LIBDCA AND NOT USE_LATE_BINDING)
+	if(USE_DCA_INSTEAD_OF_DTS)
+		add_libs_all_targets(dca)
+	else(USE_DCA_INSTEAD_OF_DTS)
 	add_libs_all_targets(dts)
+	endif(USE_DCA_INSTEAD_OF_DTS)
 endif(USE_LIBDCA AND NOT USE_LATE_BINDING)
 
 # Lame
