diff -Naur snapshot_7669.orig/avidemux/common/ADM_render/GUI_qtGlRender.cpp snapshot_7669/avidemux/common/ADM_render/GUI_qtGlRender.cpp
--- snapshot_7669.orig/avidemux/common/ADM_render/GUI_qtGlRender.cpp	2011-11-27 20:02:33.000000000 -0600
+++ snapshot_7669/avidemux/common/ADM_render/GUI_qtGlRender.cpp	2011-11-29 08:40:03.430504962 -0600
@@ -20,6 +20,7 @@
 #	define GL_TEXTURE_RECTANGLE_NV GL_TEXTURE_RECTANGLE_EXT
 #else
 #	include <GL/gl.h>
+#	include <GL/glu.h>
 #	include <GL/glext.h>
 #endif
 #define ADM_LEGACY_PROGGY // Dont clash with free/malloc etc..
@@ -401,4 +402,4 @@
 {
     return new QtGlRender();
 }
-// EOF
\ No newline at end of file
+// EOF
diff -Naur snapshot_7669.orig/avidemux/qt4/ADM_UIs/src/T_openGL.cpp snapshot_7669/avidemux/qt4/ADM_UIs/src/T_openGL.cpp
--- snapshot_7669.orig/avidemux/qt4/ADM_UIs/src/T_openGL.cpp	2011-11-27 20:02:33.000000000 -0600
+++ snapshot_7669/avidemux/qt4/ADM_UIs/src/T_openGL.cpp	2011-11-29 08:36:33.920379672 -0600
@@ -7,6 +7,7 @@
 #include "T_openGLFilter.h"
 #include "ADM_default.h"
 #include "DIA_coreToolkit.h"
+#include <GL/glu.h>
 static QGLWidget *thisWidget=NULL;
 /**
 
diff -Naur snapshot_7669.orig/avidemux_plugins/ADM_videoFilters6_openGl/glVdpau/ADM_vidVdpauFilterDeintGl.cpp snapshot_7669/avidemux_plugins/ADM_videoFilters6_openGl/glVdpau/ADM_vidVdpauFilterDeintGl.cpp
--- snapshot_7669.orig/avidemux_plugins/ADM_videoFilters6_openGl/glVdpau/ADM_vidVdpauFilterDeintGl.cpp	2011-11-27 20:02:37.000000000 -0600
+++ snapshot_7669/avidemux_plugins/ADM_videoFilters6_openGl/glVdpau/ADM_vidVdpauFilterDeintGl.cpp	2011-11-29 08:38:52.901819308 -0600
@@ -12,6 +12,7 @@
 #define GL_GLEXT_PROTOTYPES
 
 #       include <GL/gl.h>
+#       include <GL/glu.h>
 #       include <GL/glext.h>
 
 #include <QtGui/QImage> 
